src/System.Reflection/tests/AssemblyNameTests.cs

```diff
@@ -0,0 +1,377 @@
+        public void SetPublicKeyToken_GetPublicKeyToken(byte[] publicKeyToken)
+        {
+            AssemblyName assemblyName = new AssemblyName();
+            assemblyName.SetPublicKeyToken(publicKeyToken);
+            Assert.Equal(publicKeyToken, assemblyName.GetPublicKeyToken());
+        }
+        
+        [Fact]
+        public void GetPublicKeyToken_CurrentlyExecutingAssembly()
+        {
+            AssemblyName assemblyName = typeof(AssemblyNameTests).GetTypeInfo().Assembly.GetName();
+            byte[] publicKeyToken = assemblyName.GetPublicKeyToken();
+            Assert.Equal(8, publicKeyToken.Length);
+        }
+
+        [Theory]
+        [MemberData(nameof(Names_TestData))]
+        [InlineData(null, null)]
+        [InlineData("", "")]
+        public void Name_Set(string name, string expectedName)
+        {
+            AssemblyName assemblyName = new AssemblyName("MyAssemblyName");
+            assemblyName.Name = name;
+            Assert.Equal(name, assemblyName.Name);
+        }
+
+        [Fact]
+        public void Name_CurrentlyExecutingAssembly()
+        {
+            AssemblyName assemblyName = typeof(AssemblyNameTests).GetTypeInfo().Assembly.GetName();
+            Assert.True(assemblyName.Name.StartsWith("System.Reflection.Tests"));
+        }
+
+        public static IEnumerable<object[]> Version_TestData()
+        {
+            yield return new object[] { new Version(255, 1), "255.1.65535.65535" };
+            yield return new object[] { new Version(255, 1, 2), "255.1.2.65535" };
+            yield return new object[] { new Version(255, 1, 2, 3), "255.1.2.3" };
+        }
+
+        [Theory]
+        [MemberData(nameof(Version_TestData))]
+        public void Version(Version version, string versionString)
+        {
+            AssemblyName assemblyName = new AssemblyName("MyAssemblyName");
+            assemblyName.Version = version;
+
+            string expected = "MyAssemblyName, Version=" + versionString;
+            string extended = expected + ", Culture=neutral, PublicKeyToken=null";
+            Assert.True(assemblyName.FullName == expected || assemblyName.FullName == extended);
+        }
+
+        [Fact]
+        public void Version_CurrentlyExecutingAssembly()
+        {
+            AssemblyName assemblyName = typeof(AssemblyNameTests).GetTypeInfo().Assembly.GetName();
+            assemblyName.Version = new Version(255, 1, 2, 3);
+            Assert.True(assemblyName.FullName.Contains("Version=255.1.2.3"));
+        }
+
+        [Theory]
+        [InlineData("Foo")]
+        [InlineData("Hi There")]
+        public void ToString(string name)
+        {
+            var assemblyName = new AssemblyName(name);
+            Assert.True(assemblyName.ToString().StartsWith(name), string.Format("Assembly name {0} did not start with \"{1}\".", assemblyName, name));
+            Assert.Equal(assemblyName.ToString(), assemblyName.FullName);
```

Comment:
If this is a test of ToString, should the arguments here be swapped so that ToString is the actual rather than expected argument?